agent/rpc/peering/replication.go

```diff
@@ -450,13 +473,22 @@ func (s *Service) handleDelete(
 	case pbpeering.TypeURLService:
 		sn := structs.ServiceNameFromString(resourceID)
 		sn.OverridePartition(partition)
-		return s.handleUpdateService(peerName, partition, sn, nil)
+		return s.handleUpdateService(peerLocalID, peerName, partition, sn, nil)
 
 	default:
 		return fmt.Errorf("unexpected resourceURL: %s", resourceURL)
 	}
 }
 
+func servicePresentInStored(svc *structs.NodeService, stored structs.CheckServiceNodes) bool {
+	for _, csn := range stored {
+		if svc.CompoundServiceName() == csn.Service.CompoundServiceName() {
```

Comment:
Super-nit: calling `CompoundServiceName` makes allocations per iteration. Would be more efficient to capture `svc.CompoundServiceName()` in a variable to compare

Reply:
that's a good point! I'm doing away with this since we don't want to count at the service instances.